src/main/java/org/eclipsefoundation/marketplace/config/SecretConfigSource.java

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.config;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ServiceLoader;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads in a set of secret configuration values from the secret.properties file
+ * in the resources folder. These values are only secret in that they are set to
+ * be ignored by Git and should be set on a per-server basis.
+ * 
+ * ConfigSource implementation was used to enable the usage of the
+ * {@link ServiceLoader} to load the configurations.
+ * 
+ * @author Martin Lowe
+ */
+public class SecretConfigSource implements ConfigSource {
+	private static final Logger LOGGER = LoggerFactory.getLogger(SecretConfigSource.class);
+
+	private Map<String, String> secrets;
+
+	@Override
+	public Map<String, String> getProperties() {
+		if (secrets == null) {
+			this.secrets = new HashMap<>();
+			// use the class loader to get the secret.properties file path
+			URL fURL = getClass().getClassLoader().getResource("secret.properties");
+			if (fURL == null) {
+				return this.secrets;
+			}
+
+			// load the secrets file in
+			File f = new File(fURL.getFile());
+			if (!f.exists() || !f.canRead()) {
+				return this.secrets;
+			}
+
+			// read each of the lines of secret config that should be added
+			try (FileReader reader = new FileReader(f); BufferedReader br = new BufferedReader(reader)) {
```

Comment:
I guess that using the java.util.Properties class would avoid custom parsing code.

Reply:
I attempted to use Properties earlier, but whenever I would return the internal map, it wouldn't properly have read the file, and would cause issues. I can attempt to work it a little more if I can find the root cause of the issue

Reply:
It just seemed to work this time. Might be that I reused the buffered reader and it worked better than the basic file input stream I used last time.